scalableminds/webknossos
# Show Thumbnails in Dataset Table

### Description

- Introduces new design for the dashboard dataset list:
  - thumbnails replace the "type" column
  - folders and inactive datasets have icons in place of the thumbnails
  - tags are now rendered beneath the dataset name, no longer in a separate column
- Fixes a backend bug where tags with special characters (`" \ ,`) would get unexpectedly split

### Screenshots
![image](https://github.com/scalableminds/webknossos/assets/1390035/01fb4cec-7780-43d6-a17e-b6ee80802b92)

<img src="https://github.com/scalableminds/webknossos/assets/1390035/3660a65b-ead5-4fb0-a117-f1f9221ade4d" width="400" />

### URL of deployed dev instance (used for testing):
- https://datasettablethumbnails.webknossos.xyz

### Steps to test:
- Open dashboard dataset list, browse around, should look pretty
- Add and remove tags with and without special characters

### TODOs:
- [x] decide how to handle multi-layer datasets (blending? apply mapping? or select a layer?)
  - [x] include `layerNames` list in `APIDatasetCompact`
  - [x] avoid duplicated layer names (e.g. connectome_file_test_dataset) (double left join…)
  - [x] fix parsing sql array literals
  - [x] doublecheck that updating view config clears thumbnail cache for that dataset
  - [x] unify tag + name column
  - [x] add thumbnail image
  - [x] user layerNames for thumbnail layer selection
    - [x] bigger font? name is more important than tags
    - [x] tweak margins?
    - [x] styling for folders
    - [x] placeholder image for zero-layer datasets
  - [x] remove type column, have folder icon be same size as thumbnails
  - [x] repair dataset to dataset compact function
  - [x] error when entering search query (clearing localstorage helps)
  - [x] have someone who knows how to write frontend code look at it
  - [x] if no tags, add "Add Tag" label to the plus icon
  - [x] less aggressive colors for the tags
  - [x] new placeholder icons

### Follow-Up
 - #7487
 - #7490

### Issues:
- fixes #7481
- fixes #7482

------
